assign-doi-datacite/src/main/java/no/unit/nva/datacite/clients/models/ImmutableDoi.java

```diff
@@ -0,0 +1,260 @@
+package no.unit.nva.datacite.clients.models;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import nva.commons.utils.JacocoGenerated;
+
+/**
+ * Immutable implementation of {@link Doi}.
+ *
+ * <p>Use the builder to create immutable instances: {@code ImmutableDoi.builder()}.
+ */
+@SuppressWarnings({"all"})
+@javax.annotation.processing.Generated("org.immutables.processor.ProxyProcessor")
+@JacocoGenerated
+public final class ImmutableDoi extends Doi {
+
+    private static final String FORWARD_SLASH = "/";
+    private final String prefix;
+    private final String suffix;
+
+    private ImmutableDoi(String prefix, String suffix) {
+        this.prefix = prefix;
+        this.suffix = suffix;
+    }
+
+    /**
+     * Creates an immutable copy of a {@link Doi} value. Uses accessors to get values to initialize the new immutable
+     * instance. If an instance is already immutable, it is returned as is.
+     *
+     * @param instance The instance to copy
+     * @return A copied immutable Doi instance
+     */
+    public static ImmutableDoi copyOf(Doi instance) {
```

Comment:
https://github.com/BIBSYSDEV/nva-doi-registrar-client/pull/15/files/0f78749541bc60d866890e321f4aa62e89ac97d3#r521946987

Reply:
I struggle a bit with this class. It is in the `/src/main/java` folder and it has some manual additions. Should we treat it as autogenerated?  Can it be autogenerated again?

Reply:
https://immutables.github.io/immutable.html#copy-methods

It is the `Builder` who has `from` method. https://immutables.github.io/immutable.html#builder 

 I will remove the autogenerated annotations as we will maintain the generated file which is checked in since I added manually the `identifier` in the builder to set both prefix and suffix.

Reply:
Treating it as maintained code.